_posts/2018-10-08-self-arrow.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: "Desambiguating 'this' in Scala, or what the hell means `self =>`?"
+author: ragb
+---
+
+This is a quick Scala tip on a somehow unknown feature of the Scala language, what I call *self aliasing* or *unconstrained self type annotations*. One sees that in some libraries or even in the Scala standard library itself.
+It goes something like this:
+
+```scala
+trait MyTrait{self =>
+  //...
+}
+```
+
+What? You'll probably have seen a similar self type annotation type within the context of the now infamous cake pattern:
+
+```scala
+trait A {
+  def doAStuff = ???
+}
+
+trait B {this: A =>
```

Comment:
Formatting again. Give a whitespace after opening the curly bracket and probably a newline as well. @vascorsd confim this too.